SERVER-22-Refactor-And-Clean-Up-DMConnect

Danilaa56 opened the pull request:
SERVER-22-Refactor-And-Clean-Up-DMConnect
refactor: now DMHubClient read machineId file named "machineId.txt" by itself
refactor: now DMHub and DMHubClient require Start() method call and have Stop() methods
refactor: now DMHub does not disrupt Server.API application stopping

Danilaa56 committed:
fix: rename InstanceOutErr -> InstanceStdOut

Danilaa56 committed:
feat: add DisposableByteArray

Danilaa56 committed:
refactor: use disposable array in stream.ReadString, stream.ReadInt

Danilaa56 committed:
refactor: replace stream.WriteByte with stream.Write(byte[], offset, count) in stream.WriteInt

Danilaa56 committed:
refactor: move Tools/Exceptions into Domain project
feat: add AuthException
feat: add InvalidTokenException

Danilaa56 committed:
feat: add AuthResultDto for DMConnect

RemoteDedicatedMachineHub
refactor: rename -> DedicatedMachineHubClient
refactor: constructor is applied only registerDto
refactor: now HubClient stores machineId in file by itself
refactor: rework auth operation
feat: add simple "start/stop" log messages

Danilaa56 committed:
refactor: remove Action for DMConnect Dtos, use TypeNames instead of them for serialization
refactor: add clients list to hub
refactor: replace AuthOperation with AuthLoop in RemoteDedicatedMachineAgent

Danilaa56 committed:
refactor: rename RemoteDedicatedMachineAgent -> MachineAgentClient
chore: remove public keyword from interface method
chore: remove commented code files ClientAction and HubAction

Danilaa56 committed:
feat: add CancellableStreamWrapper
feat: add method Stop to DedicatedMachineHubClient, it calls Cancel on CancellationTokenSource

Danilaa56 committed:
feat: add Start/Stop methods to DedicatedMachineHub

Danilaa56 committed:
feat: add logging into DMConnect
test: add test for Auth Operation between DedicatedMachineHub and DedicatedMachineHubClient

Danilaa56 committed:
feat: add HostedHubService into Service.API. It takes care of DedicatedMachineHub
fix: now DedicatedMachineHub does not prevent stopping the Server.API application
fix: set state online on MachineAgentClient auth
fix: hub waits clients to stop on Stop() method

vekajp commented on PrivateCICDServer/DMConnect.Server.Test/Program.cs:
```diff
@@ -15,9 +19,8 @@ public DedicatedMachine RegisterMachine(RegisterDto dto)
     }
 
-    public bool AuthMachine(AuthDto dto)
+    public void AuthMachine(AuthDto dto)
```
Сюда нужно дописать что-то?

Danilaa56 replied:
Нет. Просто был либо true, либо Exception. Получается, bool не нужен

vekajp replied:
окей, тогда метод можно убирать...

Danilaa56 replied:
Там класс реализует интерфейс

vekajp commented on PrivateCICDServer/DMConnect.Client/DedicatedMachineHubClient.cs:
```diff
@@ -0,0 +1,144 @@
+        {
+            if (ExceptionUtils.IsOperationCanceled(e))
+            {
+                Console.WriteLine("Caught cancel operation");
```
Думаю, лучше это в логи писать

Danilaa56 replied:
Там, где это будет исполняться, нет логов

vekajp commented on PrivateCICDServer/DMConnect.Server.Test/Program.cs:
```diff
@@ -2,8 +2,12 @@
 
-var hub = new DedicatedMachineHub(new DedicatedMachineService(), 50050);
+var loggerFactory = LoggerFactory.Create(builder => builder.AddConsole());
+var hub = new DedicatedMachineHub(loggerFactory, new DedicatedMachineService(), 50050);
```
Порт, кажется, было бы круто вынести в config

Danilaa56 replied:
Вынес

Danilaa56 committed:
chore: fix typos in DMConnect.Client.Test

Danilaa56 committed:
feat: DMConnectHub port configuration

Danilaa56 committed:
fix: prevent collection modification exception in hub.Stop()
feat: now StreamExtension in DMConnect.Share does not require implicit ActionTypes enumeration, it finds it with reflection

Danilaa56 committed:
typo: IdDedicateMachineDto -> IsDedicateMachineDto
